### PR TITLE
Add photophore mutations

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary
```SUMMARY: Content "Add bioluminescence mutations"```

#### Purpose of change
`Weak Photophore`: A photophore protudes from your head and can glow like a glowstick for about 9kcal/224h
`Photophore`: your photophore can glow like the flashlight app of a phone for about 9kcal/112h

#### Describe the solution
Move `active_light()`to character ( #34721)
Add `lumination` property to mutations, lumination is a map of bodyparts and intensity
Check for lumination from mutations in `active_light()`
Check for coverage of affected body part()
Coverage reduces light emited by the mutation
#### Testing
Lab patient start with biolum trait
Turn on
Glow
Loose kcal
Start again with the oter trait
Glow more consume more kcal
Put on a hat -> no light
image
Drop hat -> light
image

#### Additional context
Once I move `activate_mutation()` to Character I'll probably change BIOLUM1 to either tun on randomly or turn on when the Character mood is high enough, making it a negative trait instead.

The initial idea for this trait comes from https://www.reddit.com/r/cataclysmdda/comments/dpneee/bioluminescence/

### CODE REVIEW

**Comment by irwiss** [comment]
The kcal values are weird, is it a [1kW flashlight](https://www.google.com/search?q=215kcal%2Fsec+in+watt)? Where does all the heat dissipate to?

**Comment by Zireael07** [comment]
It's not literally a flashlight, it gives off the amount of light equal to it.

File: data/json/mutations/mutations.json [hunk 1]
```diff
@@ -11,6 +11,34 @@
     "types": [ "RUNNING" ],
     "movecost_flatground_modifier": 0.85
   },
+  {
+    "type": "mutation",
+    "id": "BIOLUM1",
+    "name": "Minor bioluminescence",
```
Comment (matskuman5) [inline data/json/mutations/mutations.json]:
```suggestion
    "name": "Minor Bioluminescence",
```

**Comment by Amoebka** [comment]
Incandescent light bulbs have energy effiency of about 5% (i.e. 5% of used energy is converted into visible light). Chemical bioluminescence is about 96% efficient.

So no, the mutation should most certainly NOT consume as much energy as a flashlight to produce the same amount of light, it should consume about 20 times less.

**Comment by Fris0uman** [comment]
Considering the light has to go through the upper skin layer, your cloth ,and is produced all over your entire body to get those level of light it consume a bit more energy that a purpose build lamp.

Still the current values are not that high.

**Comment by Amoebka** [comment]
>Considering the light has to go through the upper skin layer

Why? If the intention is to mimic bioluminescence of real life animals like fireflies, realistic mechanisms and therefore numbers should be used.

You also run into the aforementioned heat issue. If the mutation consumes the same amount of energy as a flashlight and produces the same amount of light, the same 95% of energy have to be converted into heat, it can't disappear into nothing.

If you call it BIOluminescense it should actually take biology into consideration. The current number is way outside of believable ranges.

**Comment by Fris0uman** [comment]
It would be very helpful if you could provide proper numbers.

**Comment by Amoebka** [comment]
See above. 20 times lower than a flashlight of similar light output.
Assuming standart flashlight to be 1-2 watts, that means 0.04-0.08 kcal/hour.

**Comment by Fris0uman** [comment]
8kcal/100hour , since we can't track fractions of kcal, would  make the mutation free since the only way for the cost to be paid would be to keep the mutation activated  for the whole duration.

**Pull request merged**